transformations/replace_with_hyponyms_hypernyms/transformation.py

```diff
@@ -0,0 +1,132 @@
+import numpy as np
+import spacy
+import random
+from checklist.editor import Editor
+
+from interfaces.SentenceOperation import SentenceOperation
+from tasks.TaskTypes import TaskType
+
+
+class ReplaceHyponyms(SentenceOperation):
+    tasks = [TaskType.TEXT_CLASSIFICATION, TaskType.TEXT_TO_TEXT_GENERATION]
+    languages = ["en"]
+    keywords = [
+        "lexical",
+        "rule-based",
+        "external-knowledge-based",
+        "tokenizer-required",
+        "meaning-alteration",
+        "high-precision",
+        "high-generations",
+    ]
+
+    def __init__(self, n=1, seed=0, max_output=2):
+        super().__init__(seed)
+        self.nlp = spacy.load("en_core_web_sm")
+        self.n = n
+        self.max_output = max_output
+        self.editor = Editor()
+
+    def generate(self, sentence: str):
+        np.random.seed(self.seed)
+        words = []
+        perturbed_texts = []
+        tokens = self.nlp(sentence)
+        #Shuffle the tokens list so that all noun (and not just the beginning nouns) 
+        #have a fair chance at being picked.
+        shuf_tokens = list(tokens)
+        random.seed(0) #To get the same output as in test.json
+        random.shuffle(shuf_tokens)
+        for token in shuf_tokens:
+            if token.pos_ == 'NOUN':
+              words.append(token)
+              hyp_list = self.editor.hyponyms(sentence, token.text)
+              for hyp in hyp_list:
+                #Replace the noun with the hyponym
+                perturbed_texts.append(sentence.replace(token.text,hyp))
+              if len(perturbed_texts)>=self.max_output:
+                break
+        perturbed_texts = (
+            perturbed_texts[: self.max_output]
+            if len(perturbed_texts) > 0
+            else [sentence]
+        )
+        return perturbed_texts
+
+class ReplaceHypernyms(SentenceOperation):
+    tasks = [TaskType.TEXT_CLASSIFICATION, TaskType.TEXT_TO_TEXT_GENERATION]
+    languages = ["en"]
+    keywords = [
+        "lexical",
+        "rule-based",
+        "external-knowledge-based",
+        "tokenizer-required",
+        "meaning-alteration",
+        "high-precision",
+        "high-generations",
+    ]
+
+    def __init__(self, n=1, seed=0, max_output=2):
+        super().__init__(seed)
+        self.nlp = spacy.load("en_core_web_sm")
```

Comment:
You might want to use spacy like [this](https://github.com/GEM-benchmark/NL-Augmenter/blob/d39f0e5f57a79937296351996617ac03ee80fb74/transformations/close_homophones_swap/transformation.py#L55).